### PR TITLE
Implement PeeringConnection read to VPC

### PR DESCRIPTION
**Pull request by Khyme**
Implement reading peering connection and mapping the values for VPC as a datasource

closes https://github.com/timescale/terraform-provider-timescale/issues/110

part of https://github.com/timescale/forge-issues/issues/777

### CODE REVIEW

File: internal/client/queries/create_vpc.graphql
Comment (Khyme) [inline internal/client/queries/create_vpc.graphql:1]: When creating a shell vpc we didn't do peering yet

File: internal/client/vpc.go [hunk 1]
```diff
@@ -25,11 +25,11 @@ type VPC struct {
...
-	ID           string     `json:"id"`
-	VPCID        string     `json:"vpcId"`
-	Status       string     `json:"status"`
-	ErrorMessage string     `json:"errorMessage"`
-	PeerVPCs     []*PeerVPC `json:"peerVPC"`
+	ID           string   `json:"id"`
+	VPCID        string   `json:"vpcId"`
+	Status       string   `json:"status"`
+	ErrorMessage string   `json:"errorMessage"`
+	PeerVPCs     *PeerVPC `json:"peerVPC"`
```
Comment (alevy713) [inline internal/client/vpc.go:32]:
```suggestion
	PeerVPC     *PeerVPC `json:"peerVPC"`
```
does this change look right?

File: internal/provider/vpcs_data_source.go [hunk 2]
```diff
@@ -89,50 +114,41 @@ func (d *vpcsDataSource) Read(ctx context.Context, _ datasource.ReadRequest, res
...
-				VpcID:        types.Int64Value(peeringConnVpcID),
-				Status:       types.StringValue(peeringConn.Status),
-				ErrorMessage: types.StringValue(peeringConn.ErrorMessage),
-			}
-			for _, peerVpc := range peeringConn.PeerVPCs {
-				peerVpcID, err := strconv.ParseInt(peerVpc.ID, 10, 64)
-				if err != nil {
-					resp.Diagnostics.AddError("Unable to Convert Vpc ID", err.Error())
-					return
+		if len(vpc.PeeringConnections) > 0 {
```
Comment (alevy713) [inline internal/provider/vpcs_data_source.go:130]: does the state have an error without this check?

**Review by alevy713 — APPROVED** [review]

File: internal/provider/vpcs_data_source.go (hunk 2 above)
Comment (Khyme) [reply internal/provider/vpcs_data_source.go:130]: you mean should it return an error if there is no peering conn? answer to this is no, as a vpc might exist without a peering connection

Comment (alevy713) [reply internal/provider/vpcs_data_source.go:130]: sorry I meant why does it need to check if a peering connections exists. I was wondering if Terraform gives you one of those unexpected state errors if you do something this without the check

```go
var pcms []peeringConnectionDSModel                        
for _, pc := range vpc.PeeringConnections {
   ...
}
vpcState.PeeringConnections = pcms
```

Comment (Khyme) [reply internal/provider/vpcs_data_source.go:130]: Ah right, it's identical ! changing it for next PR :)

**Pull request merged**